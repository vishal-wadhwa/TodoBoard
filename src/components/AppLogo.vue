<template>
  <span :style='styling' style='border-radius:15px; padding: 2px; border-bottom: 3px double;'>
    <strong>{{mini ? 'T' : 'TODO'}}</strong>
    <em class='font-weight-thin'>Board</em>
  </span>
</template>
<script>
export default {
  props: {
    size: {
      type: [String, Number],
      default: 14
    },
    color: {
      type: String,
      default: 'white'
    },
    mini: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    styling () {
      return {
        fontSize: this.isNumericSize ? this.size + 'px' : this.size,
        color: this.color
      }
    },
    isNumericSize () {
      return !isNaN(parseFloat(this.size)) && isFinite(this.size)
    }
  }
}
</script>
