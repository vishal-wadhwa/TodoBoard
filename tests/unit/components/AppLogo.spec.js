import { shallowMount, createLocalVue, mount } from '@vue/test-utils'
import AppLogo from '@/components/AppLogo'
import Vuetify from 'vuetify'
import Vue from 'vue'
Vue.use(Vuetify)

describe('AppLogo.vue', () => {
  // required to vuetify [warn] in snapshot test
  // const app = document.createElement('div')
  // app.setAttribute('data-app', true)
  // document.body.appendChild(app)

  let localVue = null
  const factory = (opts = {}, full = false) => {
    return full ? mount(AppLogo, opts) : shallowMount(AppLogo, opts)
  }
  beforeEach(() => {
    localVue = createLocalVue()
    // localVue.use(Vuetify)
  })

  it('renders default app name', () => {
    const wrapper = factory({
      localVue
    })
    expect(wrapper.text()).toMatch('TODO Board')
  })

  it('renders shortened app name when "mini" prop is set', () => {
    const wrapper = factory({
      propsData: { mini: true },
      localVue
    })
    expect(wrapper.text()).toMatch('T Board')
  })

  it('should match snapshot with default props', () => {
    const wrapper = factory({
      localVue
    }, true)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with string size and named color', () => {
    const wrapper = factory({
      localVue,
      propsData: { size: '30rem', color: 'green' }
    }, true)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with integer size and hex color', () => {
    const wrapper = factory({
      localVue,
      propsData: { size: 30, color: '#ABCDEF' }
    }, true)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
