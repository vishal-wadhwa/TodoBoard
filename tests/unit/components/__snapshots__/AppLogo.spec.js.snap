// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLogo.vue should match snapshot with default props 1`] = `<span style="border-radius: 15px; padding: 2px; border-bottom: 3px double; font-size: 14px; color: white;"><strong>TODO</strong> <em class="font-weight-thin">Board</em></span>`;

exports[`AppLogo.vue should match snapshot with integer size and hex color 1`] = `<span style="border-radius: 15px; padding: 2px; border-bottom: 3px double; font-size: 30px; color: rgb(171, 205, 239);"><strong>TODO</strong> <em class="font-weight-thin">Board</em></span>`;

exports[`AppLogo.vue should match snapshot with string size and named color 1`] = `<span style="border-radius: 15px; padding: 2px; border-bottom: 3px double; font-size: 30rem; color: green;"><strong>TODO</strong> <em class="font-weight-thin">Board</em></span>`;
